### PR TITLE
Document Persistence Provider 

### LINKED ISSUE #1118: Storage - tangle interface code documentation
### Description
Write code comments the Tangle interface part of the storage package.

### PR DESCRIPTION
**Pull request by GalRogozinski**
# Description

Document Persistence Provider 
Fixes #1118

## Type of change
- Documentation Fix

### CODE REVIEW

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 1]
```diff
@@ -7,48 +7,190 @@
...
+     *
+     * @param model what we persist
+     * @param index the primary key that we use to store
+     * @return <tt>true</tt> if successful, else <tt>false</tt>
+     * @throws Exception if we encounter a problem with the DB
+     */
     boolean save(Persistable model, Indexable index) throws Exception;
+
+    /**
+     * Deletes from the DB the value that is stored at a specific {@code index}.
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]:
```suggestion
     * Deletes the value that is stored at a specific {@code index} from the DB.
```

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 2]
```diff
@@ -7,48 +7,190 @@
...
+     */
     void delete(Class<?> model, Indexable  index) throws Exception;
 
+    /**
+     * Updates metadata on a certain index
+     *
+     * @param model container of the metadata we need
+     * @param index the key where we store the metadata
+     * @param item extra piece of metadata
+     * @return true if successful, else false
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]:
```suggestion
     * @return <tt>true</tt> if successful, else <tt>false</tt>
```

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 3]
```diff
@@ -7,48 +7,190 @@
...
+     * @throws Exception if we encounter a problem with the DB
+     */
     boolean update(Persistable model, Indexable index, String item) throws Exception;
 
+    /**
+     * Ensures that the object of tye {@code model} is stored in the DB at {@code key}.
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]:
```suggestion
     * Ensures that the object of type {@code model} at {@code key} is stored in the DB.
```

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 4]
```diff
@@ -7,48 +7,190 @@
...
+     *
+     * @param model specifies in what table/column family to look for the object
+     * @param key the key we expect to find in the db
+     * @return true if the object is in the DB, else return false
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]:
```suggestion
     * @return <tt>true</tt> if the object is in the DB, else <tt>false</tt>
```

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 5]
```diff
@@ -7,48 +7,190 @@
...
+     * @param model specifies in what table/column family to look
+     * @param indexModel specifies the type of the key
+     * @return the (key,value) pair of the found object
+     *
+     * @throws Exception if we encounter a problem with the DB
+     */
     Pair<Indexable, Persistable> latest(Class<?> model, Class<?> indexModel) throws Exception;
 
+    /**
+     * Given 2 types of {@link Persistable} types that should be indexed with the same keys,
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]: too many "type"s

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 6]
```diff
@@ -7,48 +7,190 @@
...
+     * Retrieves a {@code model} type indexed at with {@code index} key
+     * @param model the table/column family to look at
+     * @param index the key
+     * @return The stored value
+     * @throws Exception if we encounter a problem with the DB
+     */
     Persistable get(Class<?> model, Indexable index) throws Exception;
 
+    /**
+     * Checks with a certain degree of certainity whether we have a value indexed at a
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]: too many "certain"

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 7]
```diff
@@ -7,48 +7,190 @@
...
 
+    /**
+     * Clear all the data (but not metadata) in a column family or a table
+     * @param column the table/column family we clear
+     * @throws Exception if we encounter a problem with the DB
+     */
     void clear(Class<?> column) throws Exception;
+
+    /**
+     * Clear all the metadata in  a column family or a table
```
Comment (alon-e) [inline src/main/java/com/iota/iri/storage/PersistenceProvider.java]:
```suggestion
     * Clear all the metadata in a column family or a table
```

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java [hunk 8]
```diff
@@ -7,48 +7,190 @@
 import java.util.Set;
 
 /**
- * Created by paul on 3/2/17 for iri.
+ * Abstracts access to the data persisitence layer.
+ * Will handle the initialization and connections to the storage.
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/storage/PersistenceProvider.java]: fixed

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java (hunk 5 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/storage/PersistenceProvider.java]: fixed

File: src/main/java/com/iota/iri/storage/PersistenceProvider.java (hunk 6 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/storage/PersistenceProvider.java]: fixed

**Comment by GalRogozinski** [comment]
@jakubcech 
Review when you are available please, and merge if you are fine with the changes

**Pull request merged**